lib/grape/middleware/formatter.rb

```diff
@@ -95,7 +95,7 @@ def read_rack_input(body)
         fmt = request.media_type ? mime_types[request.media_type] : options[:default_format]
 
         unless content_type_for(fmt)
-          throw :error, status: 406, message: "The requested content-type '#{request.media_type}' is not supported."
+          throw :error, status: 415, message: "The requested content-type '#{request.media_type}' is not supported."
```

Comment:
nitpicking, but I would change the message (ex. from "requested" to  "provided") to clarify the difference between 406/415

Reply:
@jdmurphy ^

Reply:
Updated, rebased and pushed as a single commit